### DISCUSSION TITLE
black screen problem when wake up from sleep

### DISCUSSION THREAD

**Discussion by WILLYWZ** [seerge/g-helper#2496] [General]
Device: ROG zephyrus M16 2023 GU604
GPU Mode/refresh rate: auto switching
When your laptop is in sleep and you remove the charger, you will likely to encounter a black screen with only the mouse. Keyboard is not working, and the only solution is to force shutdown the device. 

I've been trying to test the problem, and it is most likely due to the GPU modes. In my testing, standard mode and gpu only never encounter this problem. I don't know if the problem is from windows, nvidia optimus, or asus.

**Comment by WILLYWZ** [comment]
I think it is likely that gpu switching is triggered and your dgpu when you remove the charger (even in sleep), and when you reconnect the charger it switches again, and somehow the disabled dgpu was not successfully awaken.

**Comment by seerge** [comment]
@WILLYWZ it's extremely likely and Nvidia driver problem

They even mentioned it in latest releases
> Advanced Optimus: Notebook may not wake up from sleep when in NVIDIA GPU only display mode [4454524]

https://www.nvidia.com/download/driverResults.aspx/223498/en-us/ 

Do you use latest drivers?

**Comment by WILLYWZ** [comment]
i am using the latest studio driver. im still not sure about the problem, as i rarely use dpu only mode, and if i understand correctly, auto switch is switching between eco (igpu only) and standard (igpu+ dgpu).

**Reply by seerge** [reply]
@WILLYWZ if you refer to Nvidia's own "Auto" setting (under Nvidia Control Panel) - then it's indeed supposed to route your display to iGPU in most of the desktop scenarios and route display to dGPU when you launch game or any graphics heavy app.

**Reply by WILLYWZ** [reply]
ok. also i will try using game ready drivers to see if the problem still appears.  so do you recommend i use auto select or optimus? thanks.

**Reply by seerge** [reply]
@WILLYWZ i personally used Optimus (to ensure GPU is not being used when not needed) and setting NvidiaGPU only when i wanted to play on internal display :) 

But that's me ...

**Reply by WILLYWZ** [reply]
thanks.
